packages/ui-components/src/chat/message.tsx

```diff
@@ -192,6 +194,55 @@ export function processText(text: string): string {
     return text.replace(/\n/g, '\n\n');
 }
 
+/**
+ * A function to parse markdown such that code blocks return CodeViewer component
+ * @param message
+ */
+function parseMarkdown(message: string): React.ReactNode[] {
```

Comment:
I think we could simply utilize the tag->component map like in the example here https://github.com/remarkjs/react-markdown?tab=readme-ov-file#use-custom-components-syntax-highlight rather than relying on our own regex?